ip_country/src/country_block_stream.rs

```diff
@@ -0,0 +1,411 @@
+use csv::{StringRecord, StringRecordIter};
+use std::cmp::Ordering;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::str::FromStr;
+
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct Country {
+    pub index: usize,
+    pub iso3166: String,
+    pub name: String,
+    pub free_world: bool,
+}
+
+impl Country {
+    pub fn new(index: usize, iso3166: &str, name: &str, free_world: bool) -> Self {
+        Self {
+            index,
+            iso3166: iso3166.to_string(),
+            name: name.to_string(),
+            free_world,
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub enum IpRange {
+    V4(Ipv4Addr, Ipv4Addr),
+    V6(Ipv6Addr, Ipv6Addr),
+}
+
+impl IpRange {
+    //TODO make 2 new functions "before", "after" to check if ip is in the range or before, or after
+    // ipV4 you can represent as u32 and ipV6 as u128
+    // can reuse guts of contains for before and after and then replace arms of match witch calls to those fns
+    pub fn contains(&self, ip_addr: IpAddr) -> bool {
+        match self {
+            IpRange::V4(begin, end) => match ip_addr {
+                IpAddr::V4(candidate) => Self::contains_inner(
+                    u32::from(*begin) as u128,
+                    u32::from(*end) as u128,
+                    u32::from(candidate) as u128,
+                ),
+                IpAddr::V6(_candidate) => false,
+            },
+            IpRange::V6(begin, end) => match ip_addr {
+                IpAddr::V4(_candidate) => false,
+                IpAddr::V6(candidate) => Self::contains_inner(
+                    u128::from(*begin),
+                    u128::from(*end),
+                    u128::from(candidate),
+                ),
+            },
+        }
+    }
+
+    pub fn in_range(&self, ip_addr: IpAddr) -> Ordering {
+        match (ip_addr, self) {
+            (IpAddr::V4(ip), IpRange::V4(low, hi)) => {
+                let (low_range, hi_range) = (u32::from(*low), u32::from(*hi));
+                let ip_num = u32::from(ip);
+                if ip_num < low_range {
+                    Ordering::Greater
+                } else if ip_num > hi_range {
+                    Ordering::Less
+                } else {
+                    Ordering::Equal
+                }
+            }
+            (IpAddr::V6(ip), IpRange::V6(low, hi)) => {
+                let (low_range, hi_range) = (u128::from(*low), u128::from(*hi));
+                let ip_num = u128::from(ip);
+                if ip_num < low_range {
+                    Ordering::Greater
```

Comment:
Why do I have a feeling that these comparison operators are misused here? 
Go and check with me: 
`ip_num < low_end` can be read as "if our examined IP is smaller than the smallest IP in the range", and we should logically conclude "then the ordering is 'Less'", however you say the opposite, that is "Greater". 

I think this counter-intuitive approach should be changed towards what I depicted, it shouldn't stay as it is.